packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -263,3 +327,62 @@ export function evalBlobsAndTrees(snapshot: IOdspSnapshot) {
     }
     return { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize };
 }
+
+export function isRedeemSharingLinkError(odspResolvedUrl: IOdspResolvedUrl, error: any) {
+    const errorType = error.errorType;
+    if (odspResolvedUrl.sharingLinkToRedeem !== undefined
+        && (errorType === DriverErrorType.authorizationError
+        || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError)) {
+        return true;
+    }
+    return false;
+}
+
+function encodeShareUrl(url: string): string | undefined {
+    if (!url) {
```

Comment:
This needs to be an assert/exception, right?

Reply:
Exception was at caller place. Moved it here.